tests/functional/aws-node-sdk/test/versioning/listObjectVersions.js

```diff
@@ -1,408 +1,238 @@
+            await s3.send(new CreateBucketCommand({ Bucket: bucket }));
         });
 
-        // delete bucket after testing
-        after(done => {
-            removeAllVersions({ Bucket: bucket }, err => {
-                if (err) {
-                    return done(err);
-                }
-                return s3.deleteBucket({ Bucket: bucket }, err => {
-                    assert.strictEqual(err, null,
-                        `Error deleting bucket: ${err}`);
-                    return done();
-                });
-            });
+        after(async () => {
+            await removeAllVersions({ Bucket: bucket });
+            await s3.send(new DeleteBucketCommand({ Bucket: bucket }));
+        });
+
+        beforeEach(async () => {
+            await s3.send(new PutBucketVersioningCommand({
+                Bucket: bucket,
+                VersioningConfiguration: versioningEnabled,
+            }));
+        });
+
+        afterEach(done => {
+            removeAllVersions({ Bucket: bucket }, done);
+        });
+        
+
+        it('should list object versions', async () => {
+            const key = 'key';
+            
+            // Put first version
+            const putResult1 = await s3.send(new PutObjectCommand({
+                Bucket: bucket,
+                Key: key,
+                Body: 'body1',
+            }));
+            const versionId1 = putResult1.VersionId;
+
+            // Put second version
+            const putResult2 = await s3.send(new PutObjectCommand({
+                Bucket: bucket,
+                Key: key,
+                Body: 'body2',
+            }));
+            const versionId2 = putResult2.VersionId;
+
+            // List versions
+            const listResult = await s3.send(new ListObjectVersionsCommand({
+                Bucket: bucket,
+            }));
+
+            assert.strictEqual(listResult.Versions.length, 2);
+            
+            // Versions should be sorted by LastModified (newest first)
+            const versions = listResult.Versions.sort((a, b) => 
+                new Date(b.LastModified) - new Date(a.LastModified));
+            
+            assert.strictEqual(versions[0].VersionId, versionId2);
+            assert.strictEqual(versions[1].VersionId, versionId1);
+            assert.strictEqual(versions[0].Key, key);
+            assert.strictEqual(versions[1].Key, key);
         });
 
-        let versioning = false;
-
-        const objects = [
-            { name: 'notes/summer/august/1.txt', value: 'foo', isNull: true },
-            { name: 'notes/year.txt', value: 'foo', isNull: true },
-            { name: 'notes/yore.rs', value: 'foo', isNull: true },
-            { name: 'notes/zaphod/Beeblebrox.txt', value: 'foo', isNull: true },
-            { name: 'Pâtisserie=中文-español-English', value: 'foo' },
-            { name: 'Pâtisserie=中文-español-English', value: 'bar' },
-            { name: 'notes/spring/1.txt', value: 'qux' },
```

Comment:
are we stopping supporting object names like these? There is an advantage of keeping the special characters, as we do listing, it's to ensure it works well even when the object names are uncommon